### PR TITLE
GCP:Creation of default backing store

### PR DESCRIPTION
**Pull request by anubhav-here**
Solves #367 

Signed-off-by: Anubhav Deep <adeep@redhat.com>

### CODE REVIEW

File: pkg/system/phase4_configuring.go [hunk 1]
```diff
@@ -442,6 +442,10 @@ func (r *Reconciler) ReconcileDefaultBackingStore() error {
 		if err := r.prepareAzureBackingStore(); err != nil {
 			return err
 		}
+	} else if r.GCPCloudCreds.UID != "" {
+		log.Infof("CredentialsRequest %q created.  creating default backing store on GCP objectstore", r.GCPCloudCreds.Name)
+		if err := r.prepareGCPBackingStore(); err != nil {
```
Comment (guymguym) [inline pkg/system/phase4_configuring.go:447]: missing closing bracket maybe?

File: pkg/util/util.go [hunk 2]
```diff
@@ -807,6 +807,16 @@ func IsAzurePlatform() bool {
 	return isAzure
 }
 
+// IsGCPPlatform returns true if this cluster is running on GCP
+func IsGCPPlatform() bool {
+	nodesList := &corev1.NodeList{}
+	if ok := KubeList(nodesList); !ok || len(nodesList.Items) == 0 {
```
Comment (guymguym) [inline pkg/util/util.go:813]: It's not your PR that did that, but we ask for node list multiple times in every reconcile which seems like a noisy neighbor, so this seems like a good opportunity to refactor and move the node listing and platform detection once in the beginning of reconcile, or even just once in the process lifetime. WDYT?

File: pkg/system/phase4_configuring.go [hunk 3]
```diff
@@ -597,6 +601,27 @@ func (r *Reconciler) prepareAzureBackingStore() error {
...
+	}
+
+	util.KubeCheck(cloudCredsSecret)
+	if cloudCredsSecret.UID == "" {
+		// TODO: we need to figure out why secret is not created, and react accordingly
+		// e.g. maybe we are running on AWS but our CredentialsRequest is for Azure
+		r.Logger.Infof("Secret %q was not created yet by cloud-credentials operator. retry on next reconcile..", r.GCPCloudCreds.Spec.SecretRef.Name)
+		return fmt.Errorf("cloud credentials secret %q is not ready yet", r.GCPCloudCreds.Spec.SecretRef.Name)
+	}
+	r.Logger.Infof("Secret %s was created succesfully by cloud-credentials operator", r.GCPCloudCreds.Spec.SecretRef.Name)
```
Comment (liranmauda) [inline pkg/system/phase4_configuring.go:620]: Typo:
```suggestion
	r.Logger.Infof("Secret %s was created successfully by cloud-credentials operator", r.GCPCloudCreds.Spec.SecretRef.Name)
```

File: pkg/system/phase4_configuring.go (hunk 1 above)
Comment (anubhav-here) [reply pkg/system/phase4_configuring.go:447]: Will fix that!

File: pkg/system/phase4_configuring.go (hunk 3 above)
Comment (anubhav-here) [reply pkg/system/phase4_configuring.go:620]: Will fix :)

File: pkg/util/util.go (hunk 2 above)
Comment (anubhav-here) [reply pkg/util/util.go:813]: I mean Yeah, That sounds reasonable!

**Comment by mykaul** [comment]
@nimrod-becker - can anyone review this?

**Comment by nimrod-becker** [comment]
@mykaul it is being constantly reviewed and worked on together with @jeniawhite

**Pull request closed without merging**